WebApiCoreSeed.WebApi.Tests/Controllers/UserControllerTest.cs

```diff
@@ -0,0 +1,260 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using WebApiCoreSeed.Data.Models;
+using WebApiCoreSeed.Domain.Services.Interfaces;
+using WebApiCoreSeed.WebApi.Controllers;
+using WebApiCoreSeed.WebApi.Models;
+using Xunit;
+
+namespace WebApiCoreSeed.WebApi.Tests.Controllers
+{
+    public class UserControllerTest
+    {
+        #region Fields
+
+        private Mock<IUserService> _userService;
+
+        private UserController _classUnderTest;
+
+        #endregion
+
+        #region Constructors
+
+        public UserControllerTest()
+        {
+            _userService = new Mock<IUserService>();
+            _classUnderTest = new UserController(_userService.Object);
+        }
+
+        #endregion
+
+        #region GetAll Tests
+
+        [Fact]
+        public async void GetAll_ShouldReturnAListOfUsers_WhenHasAtLeastOneUser()
+        {
+            var id = Guid.NewGuid();
+            var users = new List<User>()
+            {
+                GetADefaultUser(id)
+            };
+
+            _userService.Setup(a => a.GetAsync()).Returns(Task.FromResult(users));
```

Comment:
You can (and this applies to all tests) use `.ReturnsAsync(users)`.

Reply:
Done, Thank you, I didn't know that method exists